src/http.rs

```diff
@@ -119,13 +300,32 @@ where
         let _enter = span.enter();
 
         default_request_headers(&mut request, &self.user_agent);
-        self.maybe_add_proxy_headers(&mut request);
+        self.maybe_add_proxy_headers(&mut request); 
 
-        emit!(http_client::AboutToSendHttpRequest { request: &request });
-
-        let response = self.client.request(request);
+        let client = self.client.clone();
+        let auth_extension = self.auth_extension.clone();
 
         let fut = async move {
+            //should request for token influence upstream service latency ?            
+            if let Some(auth_extension) = auth_extension {
+                let auth_span = tracing::info_span!("auth_extension");
+                let res = auth_extension.modify_request(&mut request)
```

Comment:
I had no idea how to handle this Box<dyn Error> to some vector friendly error better :/

Reply:
I was able to improve it a little bit with snafu 
```rust
auth_extension.modify_request(&mut request)
    .instrument(auth_span.clone().or_current())
    .await
    .inspect_err(|error| {
        // Emit the error into the internal events system.
        emit!(http_client::GotAuthExtensionError{error});
    })
    .context(ExtensionAuthenticationSnafu)?;
```